packages/cli/src/index.ts

```diff
@@ -0,0 +1,152 @@
+import { program } from "commander"
+import download from "download"
+import figlet from "figlet"
+import { existsSync, readFileSync, renameSync } from "fs"
+import logSymbols from "log-symbols"
+import { dirname } from "path"
+import { fileURLToPath } from "url"
+import Spinner from "./spinner.js"
+
+const packagePath = `${dirname(fileURLToPath(import.meta.url))}/..`
+const { description, version } = JSON.parse(readFileSync(`${packagePath}/package.json`, "utf8"))
+
+program
+    .name("semaphore")
+    .description(description)
+    .version(version, "-v, --version", "Show Semaphore CLI version.")
+    .addHelpText("before", `${figlet.textSync("Semaphore")}\n`)
+    .addHelpText("after", "\r")
+    .helpOption(undefined, "Display this help.")
+    .addHelpCommand("help [command]", "Display help for a specific command.")
+    .configureOutput({
+        outputError: (message) => {
+            console.info(`\n ${logSymbols.error}`, message)
+        }
+    })
+
+program
+    .command("init")
+    .description("Initialize a Semaphore project with a supported template.")
+    .argument("<project-directory>", "Directory of the project.")
+    // .option("-t, --template <template-name>", "Supported Semaphore template.", "hardhat")
+    .action(async (projectDirectory) => {
+        const currentDirectory = process.cwd()
+        const spinner = new Spinner(`Creating your project in ${chalk.green(`./${projectDirectory}`)}`)
+        const templateURL = `https://registry.npmjs.org/@semaphore-protocol/cli-template-hardhat/-/cli-template-hardhat-${version}.tgz`
+
+        if (existsSync(projectDirectory)) {
+            console.info(`\n ${logSymbols.error}`, `error: the '${projectDirectory}' folder already exists\n`)
+            return
+        }
+
+        spinner.start()
+
+        await download(templateURL, currentDirectory, { extract: true })
+
+        renameSync(`${currentDirectory}/package`, `${currentDirectory}/${projectDirectory}`)
+
+        spinner.stop()
+
+        console.info(` ${logSymbols.success}`, `Your project is ready!\n`)
+        console.info(` Please, install your dependencies by running:\n`)
+        console.info(`   ${chalk.cyan("cd")} ${projectDirectory}`)
+        console.info(`   ${chalk.cyan("npm i")}\n`)
+
+        const { scripts } = JSON.parse(readFileSync(`${currentDirectory}/${projectDirectory}/package.json`, "utf8"))
+
+        if (scripts) {
+            console.info(` Available scripts:\n`)
+
+            console.info(
+                `${Object.keys(scripts)
+                    .map((s) => `   ${chalk.cyan(`npm run ${s}`)}`)
+                    .join("\n")}\n`
+            )
+        }
+    })
+
+program
+    .command("get-groups")
+    .description("Get the list of groups from a supported network (Goerli or Arbitrum).")
+    .option("-n, --network <network-name>", "Supported Ethereum network.", "goerli")
+    .action(async ({ network }) => {
+        const subgraph = new Subgraph(network)
```

Comment:
Check that `network` is supported and return a nice error by the CLI instead of throwing?
```
node ../semaphore/packages/cli/dist/index.js get-groups -n Arbitrum
file:///home/r1oga/dev/semaphore/packages/subgraph/dist/index.mjs:88
            throw new TypeError("Network '".concat(network, "' is not supported"));
```

Or better, use inquirer with choices prompt
https://github.com/SBoudrias/Inquirer.js#prompt-types
that way we control what the user can select

Reply:
Thanks, I'll add a check. I tend to prefer both approaches together. For example: a user can use the optional `-n` option OR the prompt in this case.